lib/cf_deployer/stack.rb

```diff
@@ -127,8 +127,8 @@ def update_stack(template, params, capabilities, tags, override_policy_json)
       unless override_policy_json.nil?
         args[:stack_policy_during_update_body] = override_policy_json
       end
-      @cf_driver.update_stack(template, args)
-      wait_for_stack_op_terminate
+      stack_updated = @cf_driver.update_stack(template, args)
+      wait_for_stack_op_terminate if stack_updated
```

Comment:
Yes, exactly what you described.  If it failed the last time, and you aren't making any updates for this deployment (either because you're doing a dry-run, or because there are simply no changes being made from the current state after rollback), then it shouldn't count as a failed deployment.